packages/appkit/src/universal-adapter/client.ts

```diff
@@ -60,12 +63,32 @@ export class UniversalAdapter extends AdapterBlueprint {
       throw new Error('UniversalAdapter:signMessage - provider is undefined')
     }
 
-    const signature = await provider.request({
-      method: 'personal_sign',
-      params: [message, address]
-    })
+    let signature = ''
+
+    if (ChainController.state.activeCaipNetwork?.chainNamespace === ConstantsUtil.CHAIN.SOLANA) {
+      const response = await provider.request(
+        {
+          method: 'solana_signMessage',
+          params: {
+            message: bs58.encode(new TextEncoder().encode(message)),
+            pubkey: address
+          }
+        },
+        ChainController.state.activeCaipNetwork?.caipNetworkId
+      )
+
+      signature = (response as { signature: string }).signature
+    } else {
+      signature = await provider.request(
+        {
+          method: 'personal_sign',
+          params: [message, address]
+        },
+        ChainController.state.activeCaipNetwork?.caipNetworkId
+      )
+    }
 
-    return { signature: signature as `0x${string}` }
+    return { signature }
```

Comment:
the more I think about it, this method should not exist on the adapters

Reply:
I'm not sure what to do about this 😞 if we don't do it, it will break the signature for solana